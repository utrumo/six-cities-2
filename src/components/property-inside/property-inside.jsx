import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';

class PropertyInside extends PureComponent {
  render() {
    const {goods} = this.props;
    const {_renderBody} = PropertyInside;

    return goods.length ? _renderBody(goods) : null;
  }

  static _renderGoods(goods) {
    return goods.map((it, i) => (
      <li key={i} className="property__inside-item">{it}</li>
    ));
  }

  static _renderBody(goods) {
    const {_renderGoods} = PropertyInside;
    return (
      <div className="property__inside">
        <h2 className="property__inside-title">What&apos;s inside</h2>
        <ul className="property__inside-list">
          {_renderGoods(goods)}
        </ul>
      </div>
    );
  }
}

PropertyInside.propTypes = {
  goods: PropTypes.arrayOf(PropTypes.string).isRequired
};

export default PropertyInside;
