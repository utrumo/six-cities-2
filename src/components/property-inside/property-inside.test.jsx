import React from 'react';
import renderer from 'react-test-renderer';
import {PropertyInside} from './property-inside.jsx';


it(`PropertyInside should renders correctly`, () => {
  const goods = [`First`, `Second`, `Third`];
  const tree = renderer.create(<PropertyInside goods={goods} />).toJSON();
  expect(tree).toMatchSnapshot();
});
