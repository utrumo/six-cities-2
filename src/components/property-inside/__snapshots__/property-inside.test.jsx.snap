// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyInside should renders correctly 1`] = `
<div
  className="property__inside"
>
  <h2
    className="property__inside-title"
  >
    What's inside
  </h2>
  <ul
    className="property__inside-list"
  >
    <li
      className="property__inside-item"
    >
      First
    </li>
    <li
      className="property__inside-item"
    >
      Second
    </li>
    <li
      className="property__inside-item"
    >
      Third
    </li>
  </ul>
</div>
`;
